sway-core/src/type_engine/resolved_type.rs

```diff
@@ -13,29 +13,35 @@ pub enum ResolvedType {
     Unit,
     Byte,
     B256,
+    #[allow(dead_code)]
```

Comment:
What's going on with all these `dead_code`s?

Reply:
That's what I thought! The way in which I reorganized the dependencies allowed clippy to reveal that these cases were unused! We should either decide to use them, or remove them, IMO.

Reply:
Oh, wow, OK.  So, hrmm, do we think we'll need in the future?  Either way, if we do, it'll be easy to just reintroduce them, they'd not that complicated.  I vote they just go now.